### PR TITLE
Libimagstore/drop wlocks early

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This was a hard one, thanks @TheNeikos for debugging and suggestions.

@TheNeikos would be nice if you could have _really_ close look on the diff in this PR!

### CODE REVIEW

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -426,7 +428,7 @@ impl Store {
                 .map_err_into(SEK::RetrieveCallError)
         }
 
-        self.entries
+        let entry = try!(self.entries
```
Comment (TheNeikos) [inline libimagstore/src/store.rs]: This might not be enough, could you do it like this:

```rust
try!({
   /**/
});
```

Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: You mean like in acf7107 ?

Comment (TheNeikos) [inline libimagstore/src/store.rs]: Yes

**Comment by matthiasbeyer** [comment]
@TheNeikos ping.

**Comment by TheNeikos** [comment]
LGTM :+1:

**Pull request merged**